### PR TITLE
Attach original traceback in direct runner failure.

### PR DESCRIPTION
**Pull request by robertwb**
This will make it much easier to debug pipelines.

Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [ ] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

### CODE REVIEW

**Comment by robertwb** [comment]
R: @aaltay

File: sdks/python/apache_beam/runners/direct/executor.py [hunk 1]
```diff
@@ -434,9 +440,10 @@ class VisibleExecutorUpdate(object):
     raise an exception.
     """
 
-    def __init__(self, exception=None):
-      self.finished = exception is not None
-      self.exception = exception
+    def __init__(self, exc_info=(None, None, None)):
+      self.finished = exc_info[0] is not None
+      self.exception = exc_info[1] or exc_info[0]
```
Comment (aaltay) [inline sdks/python/apache_beam/runners/direct/executor.py]: What is the meaning of `exc_info[1] or exc_info[0]` ?

Comment (robertwb) [inline sdks/python/apache_beam/runners/direct/executor.py]: This translates both (ExnInstance, None, None) and (ExnType, ExnInstance, Traceback) into ExnInstance.

Comment (aaltay) [inline sdks/python/apache_beam/runners/direct/executor.py]: Thank you!

**Pull request closed without merging**